backend/core/src/Core.Application/Commands/CustomerCommands/OTPCodeByEmailCommand.cs

```
@@ -0,0 +1,79 @@
+        private readonly ICustomerDeviceRepository _customerDeviceRepository;
+        private readonly ICustomerOTPGenerator _otpGenerator;
+        private readonly ISendGridMailService _sendGridMailService;
+        private readonly IUnitOfWork _unitOfWork;
+
+        public OTPCodeByEmailCommandHandler(
+            ICustomerRepository customerRepository,
+            ICustomerDeviceRepository customerDeviceRepository,
+            ICustomerOTPGenerator otpGenerator,
+            ISendGridMailService sendGridMailService,
+            IUnitOfWork unitOfWork)
+        {
+            _customerRepository = customerRepository;
+            _customerDeviceRepository = customerDeviceRepository;
+            _otpGenerator = otpGenerator;
+            _sendGridMailService = sendGridMailService;
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task Handle(OTPCodeByEmailCommand request, CancellationToken cancellationToken)
+        {
+            var customer = await _customerRepository.GetAsync(request.CustomerCode, cancellationToken);
+
+            if (customer == null)
+            {
+                throw new CustomErrorsException(DomainErrorCode.CustomerNotFoundError.ToString(), request.CustomerCode, "Customer not found.");
+            }
+
+            if (customer.Status != CustomerStatus.ACTIVE.ToString())
+            {
+                throw new CustomErrorsException(DomainErrorCode.CustomerNotActiveError.ToString(), request.CustomerCode, "Customer is not ACTIVE.");
+            }
+
+            var customerDevice = await _customerDeviceRepository.GetAsync(request.CustomerCode, cancellationToken);
+
+            // rare case of exception
+            if (customerDevice == null || string.IsNullOrEmpty(customerDevice.OTPKey))
+            {
+                throw new CustomErrorsException(DomainErrorCode.CustomerNotFoundError.ToString(), request.CustomerCode, "Customer not found.");
```

I guess this is mostly just to hide the error? Otherwise it makes it a bit harder to figure out which of the two failures it is, since the response is identical.

---

Ideally, this should never hit. But I'll change the error message to distinguish it.